### PR TITLE
Update Magento Modules ReadMe

### PR DESCRIPTION
**Pull request by korovitskyi**
### Description (*)

Update README.md file for modules:
- Magento_Bundle;
- Magento_BundleGraphQl;
- Magento_BundleImportExport;
- Magento_CacheInvalidate;
- Magento_Captcha;
- Magento_CardinalCommerce;
- Magento_Catalog;
- Magento_CatalogCmsGraphQl;
- Magento_CatalogCustomerGraphQl;
- Magento_CatalogGraphQl.
### Fixed Issues (if relevant)

1. magento/devdocs magento/devdocs#4451: README content for modules - work in M2 Code

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/Bundle/README.md [hunk 1]
```diff
@@ -1,2 +1,126 @@
+# Magento_Bundle module
 Magento_Bundle module introduces new product type in the Magento application named Bundle Product.
+
 This module is designed to extend existing functionality of Magento_Catalog module by adding new product type.
+
+For more information about creating product, see [Creating Bundle Product](https://docs.magento.com/user-guide/catalog/product-create-bundle.html)
+
+## Installation details
+
+Before disabling or uninstalling this module, note that the following modules depends on this module:
```
Comment (rogyar) [inline app/code/Magento/Bundle/README.md]:
```suggestion
Before disabling or uninstalling this module, note that the following modules depend on this module:
```

File: app/code/Magento/Bundle/README.md [hunk 2]
```diff
@@ -1,2 +1,126 @@
...
+
+- Magento_Catalog
+- Magento_GiftMessage
+- Magento_Sales
+
+For information about module installation in Magento 2, see [Enable or disable modules](https://devdocs.magento.com/guides/v2.3/install-gde/install/cli/install-cli-subcommands-enable.html).
```
Comment (rogyar) [inline app/code/Magento/Bundle/README.md]:
```suggestion
For information about module enabling or disabling in Magento 2, see [Enable or disable modules](https://devdocs.magento.com/guides/v2.3/install-gde/install/cli/install-cli-subcommands-enable.html).
```

File: app/code/Magento/Bundle/README.md [hunk 3]
```diff
@@ -1,2 +1,126 @@
...
+ - `catalog_product_option_price_configuration_after` in the `\Magento\Bundle\Block\Catalog\Product\View\Type\Bundle::getJsonConfig()` method. Parameters:
+    - `configObj` is a config data object will be returned (`\Magento\Framework\DataObject` class).
+- `catalog_product_get_final_price` in the methods `\Magento\Bundle\Pricing\Price\BundleSelectionPrice::getValue()`, `\Magento\Bundle\Model\Product\Price::getSelectionFinalTotalPrice()`, `\Magento\Bundle\Model\Product\Price::getFinalPrice`. 
+  
+    Parameters:
+    - `product` is a product with set final price (`\Magento\Catalog\Model\Product` class).
+    - `qty` is a qty of product.
+- `prepare_catalog_product_collection_prices` in the `\Magento\Bundle\Pricing\Price\BundleSelectionPrice::getValue()` method. Parameters:
+    - `collection` is bundle collection of selection before count prices of it (`\Magento\Bundle\Model\ResourceModel\Selection\Collection` class).
+    - `store_id` is a bundle product store id.
```
Comment (rogyar) [inline app/code/Magento/Bundle/README.md]:
```suggestion
    - `store_id` is a bundle product store ID.
```

File: app/code/Magento/Bundle/README.md [hunk 4]
```diff
@@ -1,2 +1,126 @@
...
+
+ - `catalog_product_option_price_configuration_after` in the `\Magento\Bundle\Block\Catalog\Product\View\Type\Bundle::getJsonConfig()` method. Parameters:
+    - `configObj` is a config data object will be returned (`\Magento\Framework\DataObject` class).
+- `catalog_product_get_final_price` in the methods `\Magento\Bundle\Pricing\Price\BundleSelectionPrice::getValue()`, `\Magento\Bundle\Model\Product\Price::getSelectionFinalTotalPrice()`, `\Magento\Bundle\Model\Product\Price::getFinalPrice`. 
+  
+    Parameters:
+    - `product` is a product with set final price (`\Magento\Catalog\Model\Product` class).
+    - `qty` is a qty of product.
+- `prepare_catalog_product_collection_prices` in the `\Magento\Bundle\Pricing\Price\BundleSelectionPrice::getValue()` method. Parameters:
+    - `collection` is bundle collection of selection before count prices of it (`\Magento\Bundle\Model\ResourceModel\Selection\Collection` class).
```
Comment (rogyar) [inline app/code/Magento/Bundle/README.md]:
```suggestion
    - `collection` is bundle collection of selection before price calculation (`\Magento\Bundle\Model\ResourceModel\Selection\Collection` class).
```

File: app/code/Magento/Bundle/README.md [hunk 5]
```diff
@@ -1,2 +1,126 @@
...
+    - `store_id` is a bundle product store id.
+- `catalog_product_prepare_index_select` in the `\Magento\Bundle\Model\ResourceModel\Indexer\Price::prepareBundlePriceByType` method. Parameters:
+    - `select` is a select for bundle price by type to DB (`\Magento\Framework\DB\Select` class)
+    - `entity_field` is a entity id column (`\Zend_Db_Expr` class)
```
Comment (rogyar) [inline app/code/Magento/Bundle/README.md]:
```suggestion
    - `entity_field` is a entity ID column (`\Zend_Db_Expr` class)
```

File: app/code/Magento/Bundle/README.md [hunk 6]
```diff
@@ -1,2 +1,126 @@
...
+    - `website_field` is a website id column (`\Zend_Db_Expr` class)
```
Comment (rogyar) [inline app/code/Magento/Bundle/README.md]:
```suggestion
    - `website_field` is a website ID column (`\Zend_Db_Expr` class)
```

File: app/code/Magento/Bundle/README.md [hunk 7]
```diff
@@ -1,2 +1,126 @@
...
+    - `store_field` is a default store id column (`\Zend_Db_Expr` class)
```
Comment (rogyar) [inline app/code/Magento/Bundle/README.md]:
```suggestion
    - `store_field` is a default store ID column (`\Zend_Db_Expr` class)
```

File: app/code/Magento/Bundle/README.md [hunk 8]
```diff
@@ -1,2 +1,126 @@
...
+
+For information about an event in Magento 2, see [Events and observers](http://devdocs.magento.com/guides/v2.1/extension-dev-guide/events-and-observers.html#events).
```
Comment (rogyar) [inline app/code/Magento/Bundle/README.md]:
```suggestion
For information about the event system in Magento 2, see [Events and observers](http://devdocs.magento.com/guides/v2.1/extension-dev-guide/events-and-observers.html#events).
```

File: app/code/Magento/Bundle/README.md [hunk 9]
```diff
@@ -1,2 +1,126 @@
...
+    - 'widget_recently_viewed'
+
+For information about a UI component in Magento 2, see [Overview of UI components](http://devdocs.magento.com/guides/v2.1/ui_comp_guide/bk-ui_comps.html).
+
+### Public APIs
+
+`\Magento\Bundle\Api\ProductLinkManagementInterface`:
+
+- get all children for Bundle product.
+- add a child to bundle product to specified Bundle option by the product sku or product.
```
Comment (rogyar) [inline app/code/Magento/Bundle/README.md]:
```suggestion
- add a child to bundle product to specified Bundle option by the product SKU or product.
```

File: app/code/Magento/Bundle/README.md [hunk 10]
```diff
@@ -1,2 +1,126 @@
...
+    - 'widget_recently_compared'
+    - 'widget_recently_viewed'
+
+For information about a UI component in Magento 2, see [Overview of UI components](http://devdocs.magento.com/guides/v2.1/ui_comp_guide/bk-ui_comps.html).
+
+### Public APIs
+
+`\Magento\Bundle\Api\ProductLinkManagementInterface`:
+
+- get all children for Bundle product.
```
Comment (rogyar) [inline app/code/Magento/Bundle/README.md]: Please, consider using a lower-cased version of the "bundle" product everywhere (you use both versions)
```suggestion
- get all children for bundle product.
```

**Review by rogyar — CHANGES_REQUESTED** [review]
Hi @korovitskyi. Great job! Please, check my suggestions below

**Comment by korovitskyi** [comment]
Hi @rogyar.
Thanks for review.
Changes was added.

**Comment by sidolov** [comment]
@magento combine 31307 31324 31328 31335 31338 31421 31426 31429 31434

**Comment by magento-engcom-team** [comment]
Hi @sidolov. Thank you for your request. I'm working on combining the pull requests for you

**Comment by sidolov** [comment]
@magento combine 31307 31324 31328 31335 31338 31421 31426 31429 31434

**Comment by magento-engcom-team** [comment]
Hi @sidolov. Thank you for your request. I'm working on combining the pull requests for you

**Comment by magento-engcom-team** [comment]
We got some errors while combining pull requests:
 - https://github.com/magento/magento2/pull/31307 : Pull Request state must be the same for target and source pull request
 - https://github.com/magento/magento2/pull/31324 : Pull Request state must be the same for target and source pull request
 - https://github.com/magento/magento2/pull/31328 : Pull Request state must be the same for target and source pull request
 - https://github.com/magento/magento2/pull/31335 : Pull Request state must be the same for target and source pull request
 - https://github.com/magento/magento2/pull/31338 : Pull Request state must be the same for target and source pull request
 - https://github.com/magento/magento2/pull/31421 : Pull Request state must be the same for target and source pull request
 - https://github.com/magento/magento2/pull/31426 : Pull Request state must be the same for target and source pull request
 - https://github.com/magento/magento2/pull/31429 : Pull Request state must be the same for target and source pull request
 - https://github.com/magento/magento2/pull/31434 : Pull Request state must be the same for target and source pull request

**Comment by sidolov** [comment]
@magento combine 31307 31324 31328 31335

**Comment by magento-engcom-team** [comment]
Hi @sidolov. Thank you for your request. I'm working on combining the pull requests for you

**Comment by sidolov** [comment]
@magento combine 31307

**Comment by magento-engcom-team** [comment]
Hi @sidolov. Thank you for your request. I'm working on combining the pull requests for you

**Comment by magento-engcom-team** [comment]
Hi @korovitskyi
Thank you for your contribution. This pull request has been combined (merged) with https://github.com/magento/magento2/pull/31307 to optimize pull requests queue on magento/magento2 repository
Closing this pull request.

**Comment by sidolov** [comment]
@korovitskyi could you please merge all PRs related to the changes to readme.md files to one PR? 
https://github.com/magento/magento2/pull/31324
https://github.com/magento/magento2/pull/31328
https://github.com/magento/magento2/pull/31335
https://github.com/magento/magento2/pull/31338
https://github.com/magento/magento2/pull/31421
https://github.com/magento/magento2/pull/31426
https://github.com/magento/magento2/pull/31429
https://github.com/magento/magento2/pull/31434
https://github.com/magento/magento2/pull/31307

**Comment by korovitskyi** [comment]
@sidolov this PR was updated.

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Review by dobooth — CHANGES_REQUESTED** [review]
Hi @korovitskyi  Thanks for these updates. We don't bold the module names in docs. Please remove the bolding.

**Comment by korovitskyi** [comment]
Hi, @dobooth.
Thanks for the review.
Changes were added.

**Comment by engcom-Charlie** [comment]
@magento run all tests

File: app/code/Magento/Bundle/README.md [hunk 11]
```diff
@@ -1,2 +1,129 @@
...
+
+- Magento_BundleGraphQl
+- Magento_BundleImportExport
+- Magento_Wishlist
+
+For information about a module enabling or disabling in Magento 2, see [Enable or disable modules](https://devdocs.magento.com/guides/v2.4/install-gde/install/cli/install-cli-subcommands-enable.html).
+
+## Structure
+
+`Pricing/` - the directory that contain solutions for bundle product price.
```
Comment (rogyar) [inline app/code/Magento/Bundle/README.md]:
```suggestion
`Pricing/` - the directory that contains solutions for the bundle product price.
```

File: app/code/Magento/Bundle/README.md [hunk 12]
```diff
@@ -1,2 +1,129 @@
...
+### Events
+
+The module dispatches the following events:
+
+ - `catalog_product_option_price_configuration_after` in the `\Magento\Bundle\Block\Catalog\Product\View\Type\Bundle::getJsonConfig()` method. Parameters:
+    - `configObj` is a config data object will be returned(`\Magento\Framework\DataObject` class).
+- `catalog_product_get_final_price` in the methods `\Magento\Bundle\Pricing\Price\BundleSelectionPrice::getValue()`, `\Magento\Bundle\Model\Product\Price::getSelectionFinalTotalPrice()`, `\Magento\Bundle\Model\Product\Price::getFinalPrice`. 
+  
+    Parameters:
+    - `product` - a product with set final price (`\Magento\Catalog\Model\Product` class).
```
Comment (rogyar) [inline app/code/Magento/Bundle/README.md]:
```suggestion
    - `product` - a product with the final price set (`\Magento\Catalog\Model\Product` class).
```

**Review by rogyar — CHANGES_REQUESTED** [review]
Hi @korovitskyi. Just a couple of minor suggestions and we are good to go. 

Thanks!

**Comment by korovitskyi** [comment]
Hi, @yagator. 
Thanks for the review. 
Changes were added.

**Comment by ihor-sviziev** [comment]
@rogyar, could you review?

**Comment by sidolov** [comment]
@korovitskyi I'm closing the PR since the source code is not available.

**Pull request closed without merging**